klogr/klogr.go

```diff
@@ -236,35 +254,29 @@ func (l klogger) Error(err error, msg string, kvList ...interface{}) {
 	}
 }
 
-func (l klogger) V(level int) logr.Logger {
-	new := l.clone()
-	new.level = level
-	return new
-}
-
 // WithName returns a new logr.Logger with the specified name appended.  klogr
 // uses '/' characters to separate name elements.  Callers should not pass '/'
 // in the provided name string, but this library does not actually enforce that.
-func (l klogger) WithName(name string) logr.Logger {
+func (l klogger) WithName(name string) logr.LogSink {
 	new := l.clone()
```

Comment:
you should not need a clone here - taking it by value-receiver already makes a copy.  The With* methods should be copy-on-write, so this is just wasted energy.  In glogr I got rid of clone() completely

Reply:
Done!